Adding commits to clarify changes on meta-aos-rcar-gen5

longtruongrvc opened the pull request:
Dear EPAM,

I would like to inform you about the adjustments we made in the commit history, pointing out the differences between meta-aos-rcar-gen5, meta-aos-rcar-gen4, and meta-aos-vm. These repositories serve as the foundation for our ongoing porting efforts.

During the integration of AosCore modules into Gen5 VDK, we had to temporarily disable some modules related to disk encryption, such as softhsm and optee due to the limitation of VPF on disk storage, to ensure the execution of AosCore on VDK. It is part of our plan to re-enable these modules promptly as soon as the VPF is fully supported for disk storage.

For the comprehensive testing of this series, we have made a User Manual, which has been forwarded to Renesas Japan for validation. For further details, please reach out to them directly.

We appreciate your attention to this matter.

Thank you and best regards,
Long Truong.

klogg commented:
@longtruongrvc can you please abandon your previous pull request? We will review this one immediately after current release finalization

longtruongrvc commented:
Hi Artem-san,

I had closed the previous PR.

Best regards,
Long Truong.

From: Artem Mygaiev ***@***.***>
Sent: Wednesday, January 31, 2024 9:07 PM
To: aoscloud/meta-aos-rcar-gen5 ***@***.***>
Cc: Long Truong ***@***.***>; Mention ***@***.***>
Subject: Re: [aoscloud/meta-aos-rcar-gen5] Adding commits to clarify changes on meta-aos-rcar-gen5 (PR #2)


@longtruongrvc<https://github.com/longtruongrvc> can you please abandon your previous pull request? We will review this one immediately after current release finalization

-
Reply to this email directly, view it on GitHub<https://github.com/aoscloud/meta-aos-rcar-gen5/pull/2#issuecomment-1919172485>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BE3I5UDSR2VNSIBQZU6Y7EDYRJFWTAVCNFSM6AAAAABCQR2OHCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJZGE3TENBYGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

al1img commented on conf/layer.conf:
```diff
@@ -0,0 +1,12 @@
+# We have a conf and classes directory, add to BBPATH
```
We use the following 
```
[tag] Commit message
```
or
```
tag: commit message
```

klogg replied:
@longtruongrvc please check this one

longtruongrvc replied:
Ok, we will use the format "tag: Commit message" when commiting source code to the repository.

al1img commented on recipes-kernel/linux/files/aos.cfg:
```diff
@@ -0,0 +1,39 @@
+# Enable cgroups v2
```
Aos kernel configuration is not related to DTB building. Should be added as separate commit.

longtruongrvc replied:
Ok, I will correct it.

al1img commented on recipes-core/systemd/files/wait-network-override.conf:
```diff
@@ -0,0 +1,3 @@
+[Service]
```
Network online changes should be in separate commit.

longtruongrvc replied:
Ok, I will correct it.

al1img commented on recipes-core/base-files/base-files_%.bbappend:
```diff
@@ -0,0 +1,10 @@
+do_install:append() {
```
What is VPF? And why does it support only one partition?

longucci replied:
VPF is the Virtual Platform used to run Gen5 virtual device of Renesas. Currently, it is under
development so that it only allows us to use only 1 rootfs.ext4 file as rootfs.

longtruongrvc replied:
VPF stands for Virtual Platform which is used to run the Gen5 virtual device of Renesas. Currently, it is under development, which doesn't allow us to interact with the disk storage to create multiple partitions.

klogg replied:
I think that 1 partition is just a temporary solution while VPF is not fixed and/or gen5 chips are released. I would suggest moving alll the changes related to partition structure into a separate PR that will be marked as "DO NOT MERGE"

longtruongrvc replied:
I intend to address some existing limitations in the VPF:
1/ Currently, it only supports the baremetal architecture.
2/ The linux-bsp branch utilized for building the AosEdge image is not public (only accessible to Renesas employees).
3/ It currently only supports a single partition in disk storage.
4/ There is an inability to automatically reboot with reboot.service.

The majority of the code changes in this PR are aimed at overcoming these limitations. It's essential to note that these changes are temporary solutions and cannot be used for actual Gen5 chips. In my view, if this repository is intended for actual Gen5 chips, this PR should be labeled as "DO NOT MERGE".

klogg replied:
Thanks for the explanation. Are there any "generic" changes that are needed just to enable gen5 and not for the VPF? Can we somehow isolate those?

longtruongrvc replied:
We only reused the resources from meta-aos-vm and meta-aos-rcar-gen4 in two commits (9485b758c699546ab68c6b2b2770094b67f63b25 and 9c5a1220b87f6fc878af3217233a70bfe5a30987) to enable AosEdge for Gen5 chip on VPF. Other commits are workaround. I don't know is it OK to isolate these commits, could you discuss this with Azuma-san ?

klogg replied:
It is not urgent to isolate those. We can keep'em in this pull request as "do not merge" and then create a clean one for gen5 as soon as we are close to getting the HW

longtruongrvc replied:
Ok, I got it.

al1img commented on recipes-core/initrdscripts/initramfs-framework/aosupdate:
```diff
@@ -0,0 +1,262 @@
+#!/bin/sh
```
Modifying aosupdate should be done in a separate commit. First aosupdate file should be committed as is. Next commit should contain modification. Otherwise, it is hard to understand the changes done.

longtruongrvc replied:
Ok, I will correct it.

al1img commented on recipes-core/initrdscripts/initramfs-framework_%.bbappend:
```diff
@@ -0,0 +1,21 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/initramfs-framework:"
+
+SRC_URI += " \
+file://createdisk \
```
ident file:

```
RC_URI += " \
    file://createdisk \
"
```

longtruongrvc replied:
Ok, I will correct it.

al1img commented on recipes-core/initrdscripts/initramfs-framework_%.bbappend:
```diff
@@ -0,0 +1,21 @@
+"
+
+PACKAGES += " \
+initramfs-module-createdisk \
```
ditto

klogg replied:
@longtruongrvc please check this one

longtruongrvc replied:
Ok, I will correct this.

al1img commented on recipes-core/initrdscripts/initramfs-framework_%.bbappend:
```diff
@@ -0,0 +1,21 @@
+FILES:initramfs-module-createdisk = "/init.d/94-createdisk"
+
+do_install:append() {
+	install -m 0755 ${WORKDIR}/createdisk ${D}/init.d/94-createdisk
```
Use sapces for identation

longtruongrvc replied:
Ok, I will correct it.

al1img commented on recipes-core/initrdscripts/initramfs-framework/aosupdate:
```diff
@@ -0,0 +1,262 @@
+#!/bin/sh
```
Commit message: remove period at the end

longtruongrvc replied:
Ok, I will correct it.

al1img commented on recipes-core/images/aos-image-initramfs.bbappend:
```diff
@@ -0,0 +1,18 @@
+"
+
+#VPF doesn't support addtional partition
+AOS_INITRAMFS_SCRIPTS:remove = "initramfs-module-opendisk"
```
It looks like you don't need even more modules: initramfs-module-vardir, initramfs-module-machineid.
Better to specify under AOS_INITRAMFS_SCRIPTS only required modules instead of removing not required.

longtruongrvc replied:
Ok, I will correct it.

al1img commented on README.md:
```diff
@@ -1,10 +1,24 @@
+- Gather node/system information
+
+## Limitations and workarounds
+1/ No multiple partitions support for installing AosCore onto device.  
```
1.

longtruongrvc replied:
Ok, I will correct this.

al1img commented on README.md:
```diff
@@ -1,10 +1,24 @@
+- Delete initramfs-module-opendisk.
+- Add custom initramfs-module-createdisk which creates virtual disk for AOS's workdirs mountpoint.
+
+2/ Since VPF doesn't support multi-partitions storage, data within /var folder will be overwritten whenever FOTA action is performed.  
```
2.

longtruongrvc replied:
Ok, I will correct this.

al1img commented on README.md:
```diff
@@ -1,10 +1,24 @@
+Workaround:
+- Mount /var to virtual disk.
+
+3/ No reboot.service support.  
```
3.

longtruongrvc replied:
Ok, I will correct this.

al1img commented on COPYING.MIT:
```diff
@@ -0,0 +1,17 @@
+Permission is hereby granted, free of charge, to any person obtaining a copy 
```
Not relevant for this commit.

longtruongrvc replied:
Ok, I will sperate COPYING.MIT and README.md into another commit.

al1img commented on README.md:
```diff
@@ -1 +1,10 @@
-# meta-aos-rcar-gen5
\ No newline at end of file
+# meta-aos-rcar-gen5
```
ditto

longtruongrvc replied:
Ok, I will correct it.

al1img commented on fota/full.yaml:
```diff
@@ -0,0 +1,20 @@
+# Code generated by moulin. All manual changes will be lost
```
If it is autogenerated, why it is committed?

longtruongrvc replied:
Sorry for the misunderstanding, let me make this clear.
I didn't use moulin yaml file to build AosEdge image, but I based on the structure of the moulin yaml to write this full.yaml in order to create a fota_bundle for FOTA operation for Gen5 virtual device. However, I forgot to remove the comment "Code generated by moulin. All manual changes will be lost". I will correct it.

klogg replied:
Why didn'you generate it via Moulin?

longtruongrvc replied:
My team is used to using Bash script and Docker to configure and build Yocto image. We planned to export a Moulin full.yaml file to build the AosEdge image once finishing the review of this PR. However, if you want to have a Moulin file now, we will create one and push to this PR.

klogg replied:
Yes, please add to this PR so that we have a single way if building all supported products & platforms

longtruongrvc replied:
Ok, I will add the Moulin file to this PR.

al1img commented on README.md:
```diff
@@ -12,7 +12,7 @@ This release provides the following features:
 ## Limitations and workarounds
```
Could be squashed with previous commit.

longtruongrvc replied:
Ok, I will squash README.md and COPYING.MIT into 1 commit.

al1img commented on fota/full.yaml:
```diff
@@ -0,0 +1,20 @@
+# Code generated by moulin. All manual changes will be lost
+type: custom_script
```
Where is the main moulin yaml file?

longtruongrvc replied:
Since we didn't use moulin to build AosEdge image, so there is no main moulin yaml.

klogg replied:
Please use Moulin to build the image

longtruongrvc replied:
Ok, we will create a Moulin file to build the image.

longtruongrvc commented:
Hi @al1img,
Thank you for your feedback. We have carefully reviewed your comments on the pull request. We are currently in the process of addressing the points you raised. Based on your suggestions, we agree that some commit messages could benefit from separation for improved clarity. 
Would you prefer us to modify the changes and create a new PR based on your comments or would you recommend closing this current PR first and opening a new one after making the necessary modifications ?

Best regards,
Long Truong.

klogg commented:
@longtruongrvc
Please **do not** close this PR and **do not** create any new PRs. Use `git --amend` to modify / add / remove patches in this PR

P.S. There are zillions of manuals all over the internet on how to do that properly, e.g. on [StackOverflow](https://stackoverflow.com/questions/9790448/how-to-update-a-pull-request-from-forked-repo) on [GitHub](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) itself and so on. Please use standard git techniques for amending / modifying / rebasing your commit history in the PR's base branch

lehuynhlam committed:
Add layer config, README.md, license

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
Add recipes based on meta-aos-vm

Reuse from "meta-aos-vm" 9efc8e5dfdc68836a2bc088ba04843fe7c3d93ec:
meta-aos-vm/meta-aos-vm-common/recipes-support/
meta-aos-vm/meta-aos-vm-common/recipes-extended/

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
kernel: Using linux-bsp branch for Gen5 on VPF

Remove linux-bsp source and patches added in meta-renesas/meta-rcar-gateway/
Using internal branch developed for VPF, add 2 patches provided by linux-bsp team

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
kernel: Add kernel configuration for AOS

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
base-files: Modify /etc/fstab compatible with VPF Gen5

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
systemd: Only wait for tsn0 interface

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
initramfs-framework: Modify init.d/aosupdate

Due to keeping /var consistent, treat full FOTA update the same way
as incremental FOTA update

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
initramfs-framework: workaround: Add framework creating temporary disk

"createdisk" will create virtual disk for /var/aos mounting

Initramfs needs property to setup disk, separate partition into 4 smaller disks to store Aos's data
Add this to instruct "createdisk" how many disks and its size:
  recipes-aos/aos-setupdisk/aos-setupdisk.bbappend
  recipes-aos/aos-setupdisk/files/aosdisk_main.cfg

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
initramfs: workaround: Modify initramfs for Gen5 VDK

Inherit meta-aos-vm/meta-aos-vm-common/recipes-core/images/aos-image-initramfs.bbappend

Adding createdisk to initramfs
Remove incompatible "initramfs-module-opendisk", "initramfs-module-machineid"
and "initramfs-module-vardir"

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
image: Install AosCore's components and its dependencies

Get recipes-core/images/aos-image.inc
Add those packages into rcar-image-minimal.bb located in
meta-renesas/meta-rcar-gateway

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
aos: Add AosCore's config based on meta-aos-vm, meta-aos-rcar-gen4

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
aos-updatemanager: workaround:  Force UM waiting for rebooting manually

reboot.service is not supported in VPF, only shutdown and start VPF again
Adding 0001-Reboot-manuallyt.patch to modify UM code to request user's reboot trigger

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
FOTA: Add yaml to create FOTA bundle

Not support moulin yet, so modify full.yaml generated by aos-rcar-gen4.yaml

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
misc: Add unitconfig for identification on dashboard

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

longtruongrvc commented:
Hi @klogg,
We have modified the commit history based on the comments made by @al1img. Please help to review it.

Thank you and best regards,
Long Truong.

klogg commented:
Hello @longtruongrvc, I have added some major comments

longtruongrvc commented:
Hi @klogg,
Thank you for your feedback, I agree with your suggestion to isolate the temporary solution for creating a single partition that supports AosEdge on Gen5 VPF into a distinct PR marked "DO NOT MERGE".
We will proceed to adjust/rebase this PR in line with your comments.

Best regards,
Long Truong.

klogg commented:
@longtruongrvc can you please review and address all of the comments for Oleksandr and me?

longtruongrvc commented:
Hi @klogg ,
I have reviewed and addressed all the comments of you and Oleksandr. Please check it.

Best regards,
Long Truong.

klogg commented:
@longtruongrvc thanks
I have tagged you in 2 comments of Oleksandr's review that were not addressed yet, please check.

lehuynhlam committed:
refactor: Correct identations

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

lehuynhlam committed:
yaml: Add moulin yaml based on meta-aos-rcar-gen4

This moulin yaml supports AosEdge baremetal architecture only.

Signed-off-by: Lam Le <lam.le.yk@renesas.com>

longtruongrvc commented:
Hi @klogg, @al1img 
We have added 2 commits which corrects all the indentation errors within the source code and adds a Mounlin yaml file to build AosEdge image. Please help me to check it.

Best regards,
Long Truong.
